### DISCUSSION TITLE
Are there any performance tweaks for XRDP?

### DISCUSSION THREAD

**Discussion by bo-qeye** [neutrinolabs/xrdp#3000] [Q&A]
We're running xrdp 0.9.24 on Rocky Linux 8.9 as the server, and Microsoft Remote Desktop 10.9.6 (2188) on macOS as the client.

We're primarily running rather high resolutions, either 3840 × 2160 or 2560 × 1600, and my users are complaining about sluggishness when flipping pages in full screen PDF files (something we do a lot in our line of work.)

Are there any parameters that can be tweaked for a better experience? Encryption settings perhaps?

**Comment by metalefty** [comment]
Encryption is not the bottleneck at all so far.  A tweak is just a tweak and can improve performance a little bit but not significantly. There is no way to improve performance significantly in combination with xrdp v0.9 and macOS official RDP client. The current performance of v0.9 is best already if the client is a Microsoft macOS client. I know it is poor but best already at the moment. 

A mechanism called GFX is required to get better performance. Fortunately, we have already finished GFX implementation and will be out soon as xrdp v0.10.0. It is also mentioned in [the draft release note for v0.10.0](https://github.com/neutrinolabs/xrdp/wiki/NEWS-v0.10#release-notes-for-xrdp-v0100-202403dd).

v0.10.0-beta.1 is already available at the moment. You must try it if you are interested in!

**Comment by fuqifacai** [comment]
Is this a performance issue ?
If upgrade to H264 encode/decode , will it be better than now ?
